File: agbac-min/agents/java/InSessionTokenRequestTest.java
package com.agbac.tests;

import com.agbac.adapters.BaseAdapter;
import com.agbac.token.InSessionTokenRequest;
import com.agbac.sender.HybridSender;
import org.junit.jupiter.api.BeforeEach;
import org.junit.jupiter.api.Test;

import java.util.Map;

import static org.junit.jupiter.api.Assertions.*;

/**
 * Unit tests for InSessionTokenRequest.
 * Validates token requests include both sub (agent) and act (human).
 */
public class InSessionTokenRequestTest {

    private HybridSender sender;
    private BaseAdapter adapter;
    private InSessionTokenRequest tokenRequest;

    @BeforeEach
    public void setup() {
        sender = new HybridSender();
        adapter = new BaseAdapter() {
            @Override
            public String requestToken(Map<String, Object> tokenPayload) {
                // Simulated token response containing payload
                return "token:" + tokenPayload.get("client_id") + ":" + tokenPayload.get("act");
            }
        };
        tokenRequest = new InSessionTokenRequest(sender, adapter);
    }

    @Test
    public void testInSessionTokenIncludesAct() {
        Map<String, Object> sessionData = Map.of("user", Map.of("username", "alice@example.com"));
        String agentId = "agent-app";

        String token = tokenRequest.requestToken(sessionData, agentId);

        assertTrue(token.contains(agentId), "Token must contain agent client_id");
        assertTrue(token.contains("alice@example.com"), "Token must contain human act data");
    }
}

File: agbac-min/agents/typescript/tests/inSessionTokenRequest.test.ts
import { InSessionTokenRequest } from './inSessionTokenRequest';
import { HybridSender, HumanActor } from './hybridSender';
import { BaseAdapter } from './adapters/baseAdapter';

describe('InSessionTokenRequest', () => {
    it('should build token request with sub and act for in-session agent', async () => {
        const sender = new HybridSender();
        const adapter = new BaseAdapter();
        const tokenRequest = new InSessionTokenRequest(sender, adapter);

        // Mock getInSessionAct
        jest.spyOn(sender, 'getInSessionAct').mockResolvedValue({ sub: 'user:alice@example.com' });

        const payload = await tokenRequest.buildPayload();
        expect(payload).toHaveProperty('sub');
        expect(payload).toHaveProperty('act');
        expect(payload.act.sub).toBe('user:alice@example.com');
    });
});
